src/partners/0xgasless.ts

```diff
@@ -0,0 +1,250 @@
+
+type PartnerStatuses =
+  | 'other'
+  | 'created'
+  | 'completed'
+  | 'cancelled'
+  | 'payment_error'
+  | 'rejected'
+const statusMap: { [key in PartnerStatuses]: Status } = {
+  created: 'pending',
+  cancelled: 'refunded',
+  payment_error: 'refunded',
+  completed: 'complete',
+  rejected: 'refunded',
+  other: 'other'
+}
+
+export async function query0xGasless(
+  pluginParams: PluginParams
+): Promise<PluginResult> {
+  const { settings, apiKeys } = asStandardPluginParams(pluginParams)
+
+  if (apiKeys.apiKey == null) {
+    throw new Error('0xGasless: Missing 0xgasless API key')
+  }
+  const nowDate = new Date()
+  const now = nowDate.getTime()
+
+  let { latestIsoDate } = settings
+
+  if (latestIsoDate === EDGE_APP_START_DATE) {
+    latestIsoDate = new Date(PLUGIN_START_DATE).toISOString()
+  }
+
+  let lastCheckedTimestamp = new Date(latestIsoDate).getTime() - QUERY_LOOKBACK
+  if (lastCheckedTimestamp < 0) lastCheckedTimestamp = 0
+
+  const ssFormatTxs: StandardTx[] = []
+  let retry = 0
+
+  while (true) {
+    const startTimestamp = lastCheckedTimestamp
+    const endTimestamp = lastCheckedTimestamp + QUERY_TIME_BLOCK_MS
+
+    try {
+      let cursor: string | undefined
+
+      while (true) {
+        const urlObj = new URL(API_URL, true)
+
+        const queryParams: {
+          startTimestamp: string
+          endTimestamp: string
+          cursor?: string
+        } = {
+          // API expects seconds-based unix timestamps
+          startTimestamp: Math.floor(startTimestamp / 1000).toString(),
+          endTimestamp: Math.floor(endTimestamp / 1000).toString()
+        }
+        if (cursor != null) queryParams.cursor = cursor
+        urlObj.set('query', queryParams)
+
+        datelog(
+          `0xGasless Querying from:${new Date(
+            startTimestamp
+          ).toISOString()} to:${new Date(endTimestamp).toISOString()}`
+        )
+
+        const url = urlObj.href
+        const response = await retryFetch(url, {
+          headers: {
+            '0x-api-key': apiKeys.apiKey,
+            '0x-version': 'v2'
+          }
+        })
+        const responseJson = await response.text()
+        if (!response.ok) {
+          throw new Error(`${url} response ${response.status}: ${responseJson}`)
+        }
+        const responseBody = asGetGaslessTradesResponse(responseJson)
+
+        for (const trade of responseBody.trades) {
+          const buySymbol = trade.tokens.find(t => t.address === trade.buyToken)
+            ?.symbol
+          const sellSymbol = trade.tokens.find(
+            t => t.address === trade.sellToken
+          )?.symbol
+
+          if (buySymbol == null || sellSymbol == null) {
+            throw new Error(
+              `Could not find buy or sell symbol for trade ${trade.zid}; txid: ${trade.transactionHash}`
+            )
+          }
+
+          const {
+            isoDate: tradeIsoDate,
+            timestamp: tradeTimestamp
```

Comment:
why rename `isoDate` and `timestamp`. There's no conflict and it would match the StandardTx type for easy assignment